client/ayon_deadline/repository/custom/plugins/GlobalJobPreLoad.py

```diff
@@ -651,6 +574,142 @@ def inject_ayon_environment(deadlinePlugin):
+    Raises:
+        (RuntimeError) if extraction takes more
+            than EXTRACT_ENVIRONMENT_TIMEOUT seconds
+    """
+    start_time = datetime.now()
+    while os.path.exists(f"{export_url}.tmp"):
+        date_diff = datetime.now() - start_time
+        if date_diff > timedelta(seconds=EXTRACT_ENVIRONMENT_TIMEOUT):
+            raise RuntimeError(
+                "Previous extract environment process stuck for "
+                f"'{EXTRACT_ENVIRONMENT_TIMEOUT}' sec."
+                "Starting it from scratch."
+            )
+        print("Extract environment process already triggered, waiting")
+        sleep(2)
+
+
+def _get_output_dir(job):
+    """Look for output dir where metadata.json should be created also."""
+    output_urls = ["OutputFilePath", "Output", "SceneFile"]
+    output_dir = None
+    for output in output_urls:
+        output_dir = job.GetJobPluginInfoKeyValue(output)
+        if output_dir:
+            break
+    if not output_dir:
+        raise RuntimeError(
+            "Unable to find workfile location or "
+            "location where files should be rendered.")
+    output_dir = os.path.dirname(output_dir)
+    return output_dir
+
+
+def _extractenvironments(
+    ayon_server_url,
+    ayon_api_key,
+    ayon_bundle_name,
+    deadlinePlugin,
+    exe,
+    export_url,
+    job
+):
+    """Calls `applications.extractenvironments` cli to get farm based envs."""
+    print(f">>> Extracting environments to: {export_url}")
+
+    add_kwargs = {
+        "envgroup": "farm",
+    }
+    # Support backwards compatible keys
+    for key, env_keys in (
+            ("project", ["AYON_PROJECT_NAME", "AVALON_PROJECT"]),
+            ("folder", ["AYON_FOLDER_PATH", "AVALON_ASSET"]),
+            ("task", ["AYON_TASK_NAME", "AVALON_TASK"]),
+            ("app", ["AYON_APP_NAME", "AVALON_APP_NAME"]),
+    ):
+        value = ""
+        for env_key in env_keys:
+            value = job.GetJobEnvironmentKeyValue(env_key)
+            if value:
+                break
+        add_kwargs[key] = value
+    if not all(add_kwargs.values()):
+        raise RuntimeError((
+            "Missing required env vars: AYON_PROJECT_NAME,"
+            " AYON_FOLDER_PATH, AYON_TASK_NAME, AYON_APP_NAME"
+        ))
+    # Use applications addon arguments
+    # TODO validate if applications addon should be used
+    args = [
+        "--headless",
+        "addon",
+        "applications",
+        "extractenvironments",
+        export_url
+    ]
+    # staging requires passing argument
+    # TODO could be removed when PR in ayon-core starts to fill
+    #  'AYON_USE_STAGING' (https://github.com/ynput/ayon-core/pull/1130)
+    #  - add requirement for "core>=1.1.1" to 'package.py' when removed
+    settings_variant = job.GetJobEnvironmentKeyValue(
+        "AYON_DEFAULT_SETTINGS_VARIANT"
+    )
+    if settings_variant == "staging":
+        args.append("--use-staging")
+    # Backwards compatibility for older versions
```

Comment:
We should maybe get rid of the legacy handling. It is from time before ayon-core 1.0.0 release.

Reply:
Removed.